common/src/main/java/com/wynntils/features/SilencerFeature.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright © Wynntils 2023.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.features;
+
+import com.wynntils.core.components.Managers;
+import com.wynntils.core.components.Models;
+import com.wynntils.core.config.Category;
+import com.wynntils.core.config.Config;
+import com.wynntils.core.config.ConfigCategory;
+import com.wynntils.core.config.RegisterConfig;
+import com.wynntils.core.features.Feature;
+import com.wynntils.core.features.properties.RegisterKeyBind;
+import com.wynntils.core.keybinds.KeyBind;
+import com.wynntils.core.mod.event.WynncraftConnectionEvent;
+import com.wynntils.core.storage.Storage;
+import com.wynntils.mc.event.TitleScreenInitEvent;
+import com.wynntils.utils.mc.McUtils;
+import net.minecraft.ChatFormatting;
+import net.minecraft.network.chat.Component;
+import net.minecraft.sounds.SoundSource;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+import org.lwjgl.glfw.GLFW;
+
+@ConfigCategory(Category.UNCATEGORIZED)
+public class SilencerFeature extends Feature {
+    @RegisterConfig
+    public final Config<Double> silencerVolume = new Config<>(0.01);
+
+    @RegisterKeyBind
+    private final KeyBind silencerKeyBind =
+            new KeyBind("Toggle Silencer", GLFW.GLFW_KEY_UNKNOWN, true, this::toggleSilencer);
+
+    private final Storage<Double> lastVolume = new Storage<>(1.0);
```

Comment:
The decision to reset the volume is based on my design choice to enhance user-friendliness. This feature is intended as a "session-only" setting, typically used for grinding sessions or multitasking. I see two primary user scenarios:

1. Users finish their task, decide to do something different on the server, notice their sound is "silenced", and remember to disable the Silencer themselves.
2. Users log off and/or close the game by any means and return later, possibly on a different day.

In the second scenario, we reset the volume to the their value. It's unrealistic to expect users to remember the last Silencer state between different gaming sessions. To facilitate resetting on both disconnect and in subsequent Minecraft runs, we need `lastVolume` to be a storage.

Reply:
But the volume is a vanilla option setting that is persisted between runs, yes? I'm not sure I follow the logic here. If we start the silencer, the volume will go from lets say 100 to 1. This is stored in vanillas option, and we store the 100 in our lastVolume. Then the user quits. Or the game crashes, or whatever.

When they restart, the vanilla volume will be 1. But if they log on to Wynncraft, Wynntils will think the silencer is disabled. So if they enable the silencer, it will store 1 as the lastVolume, and set the volume to 0.01.

This seems broken.

If this is not the case, what am I missing?

Reply:
Yes you understood correctly but you missed that we attempt to `resetVolume` during `onDisconnect` and additionally we also "force" this during "launch" `onTitleScreenInit`.

Therefore, instead of the volume transitioning from 100 to 1 and then to 0.01, the process looks more like this:

1. We toggle Silencer to 'on'
2. The volume goes from 100 to 1
3. We store 100 in `lastVolume`; Vanilla settings store 1
4. Upon a crash, exit, etc., we **attempt** to toggle Silencer to 'off' and we try to load Volume 100 from `lastVolume`
5. Upon reboot, we force Silencer to 'off' note that this is because of `private boolean isSilencerEnabled = false;` is the starting state
7. Vanilla settings load with Volume at 1
8. Vanilla settings attempt to load the Title Screen
9. We load Volume 100 from `lastVolume`
10. Vanilla settings finally load the Title Screen and we can proceed with `autoConnect` or nothing.